main.go

```diff
@@ -1,15 +1,73 @@
 package main
 
 import (
+	"database/sql"
+	"encoding/json"
 	"flag"
-	"log"
+	"fmt"
+	"os"
+
+	_ "github.com/lib/pq"
+
 	"net/http"
 	"strings"
 
 	"github.com/gorilla/websocket"
+	log "github.com/sirupsen/logrus"
 )
 
-var addr = flag.String("addr", "localhost:8080", "http service address")
+// blockStats is the information to report about individual blocks.
+type Block struct {
+	Number     int       `json:"number"`
+	Hash       string    `json:"hash"`
+	ParentHash string    `json:"parentHash"`
+	Timestamp  int       `json:"timestamp"`
+	Miner      string    `json:"miner"`
+	GasUsed    uint64    `json:"gasUsed"`
+	GasLimit   uint64    `json:"gasLimit"`
+	Diff       string    `json:"difficulty"`
+	TotalDiff  string    `json:"totalDifficulty"`
+	Txs        []TxStats `json:"transactions"`
+	TxHash     string    `json:"transactionsRoot"`
+	Root       string    `json:"stateRoot"`
+	Uncles     []Block   `json:"uncles"`
+}
+
+type TxStats struct {
+	Hash string `json:"hash"`
+}
+
+// nodeInfo is the collection of meta information about a node that is displayed
+// on the monitoring page.
+type NodeInfo struct {
+	Name     string `json:"name"`
+	Node     string `json:"node"`
+	Port     int    `json:"port"`
+	Network  string `json:"net"`
+	Protocol string `json:"protocol"`
+	API      string `json:"api"`
+	Os       string `json:"os"`
+	OsVer    string `json:"os_v"`
+	Client   string `json:"client"`
+	History  bool   `json:"canUpdateHistory"`
+}
+
+var rawInfo NodeInfo
```

Comment:
You cannot do this, why do you have a global variable that you are going to decode on every websocket request? This is not going to work when multiple nodes send metrics and there is only one var to decode the object. Move this to the echo function.